### PR TITLE
Add Retired sidebar

### PR DESCRIPTION
**Pull request by weshaggard**

### CODE REVIEW

File: _data/sidebars/releases_sidebar.yml [hunk 1]
```diff
@@ -97,3 +97,14 @@ entries:
...
+      url: /releases/retired/index.html
+      folderitems:
+      - title: .NET
+        url: /releases/retired/dotnet.html
+      - title: Java
+        url: /releases/retired/java.html
+      - title: JavaScript/TypeScript
+        url: /releases/retired/js.html
+      - title: Python
+        url: /releases/retired/python.html
```
Comment (mario-guerra) [inline _data/sidebars/releases_sidebar.yml:110]: Does it make sense to add a link for Go now or should we wait till we actually retire a Go library?

Comment (mario-guerra) [reply _data/sidebars/releases_sidebar.yml:110]: After thinking about it a bit, I think we should add a link for Go now. There are active Go users and they may want to use this page to check if they are using the most recent libraries.

Comment (weshaggard) [reply _data/sidebars/releases_sidebar.yml:110]: I would wait until we have something there.

Comment (mario-guerra) [reply _data/sidebars/releases_sidebar.yml:110]: Okay, sounds good.

Comment (weshaggard) [inline _data/sidebars/releases_sidebar.yml:110]:
```suggestion
    url: /releases/retired/index.html
    folderitems:
    - title: .NET
      url: /releases/retired/dotnet.html
    - title: Java
      url: /releases/retired/java.html
    - title: JavaScript/TypeScript
      url: /releases/retired/js.html
    - title: Python
      url: /releases/retired/python.html
```

**Pull request merged**